server/src/main/java/io/druid/server/security/AnonymousAuthenticator.java

```diff
@@ -0,0 +1,130 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.server.security;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonTypeName;
+
+import javax.servlet.DispatcherType;
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import java.io.IOException;
+import java.util.EnumSet;
+import java.util.Map;
+
+/**
+ * Authenticates all requests and directs them to an authorizer.
+ */
+@JsonTypeName("anonymous")
+public class AnonymousAuthenticator implements Authenticator
+{
+  private final AuthenticationResult anonymousResult;
+  private final String name;
+  private final String authorizerName;
+  private final String identity;
+  private final String DEFAULT_IDENTITY = "defaultUser";
+
+  @JsonCreator
+  public AnonymousAuthenticator(
+      @JsonProperty("name") String name,
+      @JsonProperty("authorizerName") String authorizerName,
+      @JsonProperty("identity") String identity
+  )
+  {
+    this.name = name;
+    this.authorizerName = authorizerName;
+    this.identity = identity == null ? DEFAULT_IDENTITY : identity;
+    this.anonymousResult = new AuthenticationResult(
```

Comment:
nit : It seems only the anonymousResult needs to be a field in the class, rest can be removed ? 
In case you meant this class to be used in JSON serde, please add @JsonProperty annotations to the fields and create their getter methods.

Reply:
Hi @nishantmonu51, thanks for your comment - I've removed those other fields. Does this look ok to you?